tests/test_bookmark.py

```diff
@@ -1,109 +1,125 @@
-        custom_id = self.custom_reports_names_to_ids()['Test Report 1']
         return {
             'content_group_report',
-            custom_id
+            'Test Report 1'
         }
 
+    bookmark_format = "%Y-%m-%d"
+    initial_bookmarks = {}
 
-    def manipulate_state(self, old_state):
-        manipulated_state = {
-            'bookmarks': {
-                stream_id: { os.getenv('TAP_GA4_PROPERTY_ID'): {'last_report_date': self.bookmark_date}}
-                for stream_id in old_state['bookmarks'].keys()
-            }
-        }
+    @staticmethod
+    def name():
+        return "tt_ga4_bookmark"
 
-        return manipulated_state
+    def manipulate_state(self, state: dict, new_bookmarks: dict):
+        """
+        This method will update the passed state with the new_bookmarks.
 
+        new_bookmarks must be in the format { stream: {replication_key: replication_value}}
+        """
+        new_state = deepcopy(state)
+        if new_state.get('bookmarks') is None:
+            new_state['bookmarks'] = {}
+        for stream, rep in new_bookmarks.items():
+            if new_state['bookmarks'].get(stream):
+                for value in rep.values():
+                    new_state['bookmarks'][stream][os.getenv('TAP_GA4_PROPERTY_ID')] = \
+                        {'last_report_date': value}
+            else:
+                # It is expected there will be only one key value pair.
+                # if that is not the case this will only use the last value in the key value pair
+                for value in rep.values():
+                    new_state['bookmarks'][stream] = {
+                        os.getenv('TAP_GA4_PROPERTY_ID'): {'last_report_date': value}}
+        return new_state
+
+    def get_bookmark_value(self, state, stream):
+        bookmark = state.get('bookmarks', {})
+        stream_bookmark = bookmark.get(self.get_stream_id(stream))
+        if stream_bookmark:
+            return stream_bookmark.get(os.getenv('TAP_GA4_PROPERTY_ID')).get('last_report_date')
+        return None
 
-    def streams_to_selected_fields(self):
+    @staticmethod
+    def streams_to_selected_fields():
+        # TODO - when selecting these fields I'm getting errors.
         return {
             "Test Report 1": {
                 "conversions",
-                "defaultChannelGrouping",
-                "eventName",
-                "eventCount",
-                "newUsers",
-                "enagementRate",
-                "engagedSessions",
-            },
-            'content_group_report': {
-                "date",
-                "browser",
-                "conversions",
+                # "defaultChannelGrouping",  # Are these are valid selections?
+                # "eventName",
+                # "eventCount",
+                # "newUsers",
+                # "enagementRate",
+                # "engagedSessions",
             },
+            'content_group_report': set()
+            #     {
```

Comment:
content group report can only have these fields in it: https://github.com/singer-io/tap-ga4/blob/main/tap_ga4/reports.py#L320-L332 
i'm not sure when browser was a valid field for this report